pallets/prices/src/lib.rs

```diff
@@ -154,6 +160,19 @@ pub mod pallet {
             <Pallet<T> as EmergencyPriceFeeder<CurrencyId, Price>>::reset_emergency_price(asset_id);
             Ok(().into())
         }
+
+        /// Set foreign vault token mapping
+        #[pallet::weight((<T as Config>::WeightInfo::set_foreign_vault_mapping(), DispatchClass::Operational))]
+        #[transactional]
+        pub fn set_foreign_vault_mapping(
+            origin: OriginFor<T>,
+            foreign_asset_id: CurrencyId,
+            asset_id: CurrencyId,
+        ) -> DispatchResultWithPostInfo {
+            T::FeederOrigin::ensure_origin(origin)?;
```

Comment:
The FeederOrigin naming looks wierd in this case

Reply:
how about just use sudo here